pkg/cloud/gcp/actuators/util/gcp_uefi_disk_check.go

```diff
@@ -0,0 +1,131 @@
+
+		if imageRef.IsFamily {
+			return checkImageFamilyImageUEFICompatible(gceService, imageRef.Project, providerConfig.Zone, imageRef.Image)
+		}
+		return checkImageUEFICompatible(gceService, imageRef.Project, imageRef.Image)
+	}
+	return false, fmt.Errorf("no boot disk found")
+}
+
+// imageReference holds parsed details from a disk.Image string.
+type imageReference struct {
+	Project  string
+	Image    string
+	IsFamily bool // true if the image is specified as a family reference.
+}
+
+// parseImageReference extracts project and image information from the given image string.
+// It supports various formats:
+//   - "projects/{project}/global/images/{image}"
+//   - "projects/{project}/global/images/family/{imageFamily}"
+//   - A simple image name without slashes, in which case providerProject is used.
+func parseImageReference(imageStr, providerProject string) (*imageReference, error) {
+	// If the image string does not contain a slash, assume it's a simple image name.
+	if !strings.Contains(imageStr, "/") {
+		return &imageReference{
+			Project:  providerProject,
+			Image:    imageStr,
+			IsFamily: false,
+		}, nil
+	}
+
+	// Check if the image string contains "projects/".
+	if !strings.Contains(imageStr, "projects/") {
+		return nil, fmt.Errorf("image string %q does not contain expected 'projects/' segment", imageStr)
+	}
+
+	// Split based on "projects/".
+	parts := strings.SplitN(imageStr, "projects/", 2)
+	if len(parts) < 2 {
+		return nil, fmt.Errorf("unexpected format for image string: %q", imageStr)
+	}
+
+	// Split the remainder by "/".
+	subParts := strings.Split(parts[1], "/")
+	// Expected formats:
+	// For non-family images: {project}/global/images/{image} => at least 4 parts.
+	// For family images: {project}/global/images/family/{imageFamily} => at least 5 parts.
+	if len(subParts) < 4 {
+		return nil, fmt.Errorf("unexpected image path format in %q", imageStr)
+	}
+
+	// Determine if the image is specified as a family.
+	isFamily := false
+	imageName := ""
+	if len(subParts) >= 5 && subParts[2] == "images" && subParts[3] == "family" {
+		isFamily = true
+		imageName = subParts[4]
+	} else if subParts[1] == "global" && subParts[2] == "images" {
+		imageName = subParts[3]
+	} else {
+		return nil, fmt.Errorf("unrecognized image path format in %q", imageStr)
+	}
+
+	return &imageReference{
+		Project:  subParts[0],
+		Image:    imageName,
+		IsFamily: isFamily,
+	}, nil
+}
+
+// checkImageUEFICompatible retrieves the image and checks its GuestOSFeatures for UEFI support.
+func checkImageUEFICompatible(gceService computeservice.GCPComputeService, project, image string) (bool, error) {
+	img, err := gceService.ImageGet(project, image)
+	if err != nil {
+		return false, fmt.Errorf("unable to retrieve image %q in project %q: %w", image, project, err)
+	}
+	for _, feat := range img.GuestOsFeatures {
+		if strings.Contains(feat.Type, UEFICompatible) {
+			klog.Infof("Image %q is UEFI compatible: %v", image, img.GuestOsFeatures)
```

Comment:
At how many layers do we log this? I thought I saw a similar log at the call site 🤔

Reply:
Yep, makes sense - there were debuggy I left in. 

Have moved up, and changed the log that the check is beginning to be only visible at `v=3` as I think it'll be noise otherwise.